src/unittests/f32_interval.spec.ts

```diff
@@ -0,0 +1,439 @@
+      { value: hexToF64(0x00000000, 0x00000002), error: 2 ** -11, expected: [-(2 ** -11), 2 ** -11] },
+      { value: hexToF64(0x00000000, 0x00000002), error: 1, expected: [-1, 1] },
+      { value: hexToF64(0x800fffff, 0xffffffff), error: 0, expected: [kValue.f32.subnormal.negative.max, 0] },
+      { value: hexToF64(0x800fffff, 0xffffffff), error: 2 ** -11, expected: [-(2 ** -11), 2 ** -11] },
+      { value: hexToF64(0x800fffff, 0xffffffff), error: 1, expected: [-1, 1] },
+      { value: hexToF64(0x800fffff, 0xfffffffe), error: 0, expected: [kValue.f32.subnormal.negative.max, 0] },
+      { value: hexToF64(0x800fffff, 0xfffffffe), error: 2 ** -11, expected: [-(2 ** -11), 2 ** -11] },
+      { value: hexToF64(0x800fffff, 0xfffffffe), error: 1, expected: [-1, 1] },
+
+      // Zero
+      { value: 0, error: 0, expected: [0, 0] },
+      { value: 0, error: 2 ** -11, expected: [-(2 ** -11), 2 ** -11] },
+      { value: 0, error: 1, expected: [-1, 1] },
+    ]
+  )
+  .fn(t => {
+    const value = t.params.value;
+    const error = t.params.error;
+    const expected = arrayToInterval(t.params.expected);
+
+    const got = absoluteErrorInterval(value, error);
+    t.expect(
+      objectEquals(expected, got),
+      `absoluteErrorInterval(${value}, ${error}) returned ${got}. Expected ${expected}`
+    );
+  });
+
+interface ULPCase {
+  value: number;
+  num_ulp: number;
+  expected: [number, number];
+}
+
+g.test('ulpInterval')
+  .paramsSubcasesOnly<ULPCase>(
+    // prettier-ignore
+    [
+      // Edge Cases
+      { value: kValue.f32.infinity.positive, num_ulp: 0, expected: [kValue.f32.positive.max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.infinity.positive, num_ulp: 1, expected: [kValue.f32.positive.max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.infinity.positive, num_ulp: 4096, expected: [kValue.f32.positive.max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.infinity.negative, num_ulp: 0, expected: [Number.NEGATIVE_INFINITY, kValue.f32.negative.min] },
+      { value: kValue.f32.infinity.negative, num_ulp: 1, expected: [Number.NEGATIVE_INFINITY, kValue.f32.negative.min] },
+      { value: kValue.f32.infinity.negative, num_ulp: 4096, expected: [Number.NEGATIVE_INFINITY, kValue.f32.negative.min] },
+      { value: kValue.f32.positive.max, num_ulp: 0, expected: [kValue.f32.positive.max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.positive.max, num_ulp: 1, expected: [kValue.f32.positive.nearest_max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.positive.max, num_ulp: 4096, expected: [hexToF32(0x7f7fefff), Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.positive.min, num_ulp: 0, expected: [ kValue.f32.positive.min,  kValue.f32.positive.min] },
+      { value: kValue.f32.positive.min, num_ulp: 1, expected: [0, hexToF32(0x01000000)] },
+      { value: kValue.f32.positive.min, num_ulp: 4096, expected: [hexToF32(0x867ff000), hexToF32(0x06800800)] },
```

Comment:
`kValue.f32.positive.min` (`0x00800000`) is: `1`  ×  <code>2<sup>-126</sup></code>  ×  `1`	
`0x867ff000` is:  `-1`  ×  <code>2<sup>-115</sup></code>  ×  `1.9995117`
`0x06800800` is:  `1`  ×  <code>2<sup>-114</sup></code>  ×  `1.0002441`

I get that the ULP could skip subnormals, making the low range negative, but the exponents here seem off?

Given that we're dealing with normal values here, wouldn't the high-range of this be `0x00801000` (`0x00800000` + `4096`)?

I could (and probably am) just missing something.